src/Microsoft.AspNetCore.Mvc.Core/Routing/ControllerLinkGeneratorExtensions.cs

```diff
@@ -0,0 +1,282 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Mvc.Routing;
+using System;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    /// <summary>
+    /// Extension methods for using <see cref="LinkGenerator"/> to generate links to MVC controllers.
+    /// </summary>
+    public static class ControllerLinkGeneratorExtensions
+    {
+        private static readonly LinkGenerationTemplateOptions _templateOptions = new LinkGenerationTemplateOptions()
+        {
+            UseAmbientValues = true,
+        };
+
+        /// <summary>
+        /// Generates a URI with an absolute path based on the provided values.
+        /// </summary>
+        /// <param name="generator">The <see cref="LinkGenerator"/>.</param>
+        /// <param name="httpContext">The <see cref="HttpContext"/> associated with the current request.</param>
+        /// <param name="action">
+        /// The action name. Used to resolve endpoints. Optional. If <c>null</c> is provided, the current action route value
+        /// will be used.
+        /// </param>
+        /// <param name="controller">
+        /// The controller name. Used to resolve endpoints. Optional. If <c>null</c> is provided, the current controller route value
+        /// will be used.
+        /// </param>
+        /// <param name="values">The route values. Optional. Used to resolve endpoints and expand parameters in the route template. Optional.</param>
+        /// <param name="pathBase">
+        /// An optional URI path base. Prepended to the path in the resulting URI. If not provided, the value of <see cref="HttpRequest.PathBase"/> will be used.
```

Comment:
nit: Some params start with "an optional XXX" and others are all like "The XXX. Optional."

Pick one for consistency?

Reply:
Sure, I'll do the latter.